spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java

```diff
@@ -174,12 +175,39 @@ private <T> List<Entity> getEntitiesForSave(
     return entitiesForSave;
   }
 
+  @Override
+  public <T> T insert(final T instance, final Key... ancestors) {
```

Comment:
Instead of duplicating the logic of `save()` and `saveAll()`, can you add a private method that can be used by both? It would just need an additional argument `Consumer<FullEntity<?>[]> consumer`.